### PR TITLE
gh-110774: allow setting the Runner(loop_factory=...) from IsolatedAsyncioTestCase

### PR DESCRIPTION
**Pull request by graingert**
* Issue: gh-110774

### CODE REVIEW

File: Doc/library/unittest.rst [hunk 1]
```diff
@@ -92,6 +92,8 @@ Here is a short script to test three string methods::
 
   class TestStringMethods(unittest.TestCase):
 
+      loop_factory = asyncio.EventLoop
+
```
Comment (graingert) [inline Doc/library/unittest.rst]: wrong example
```suggestion
```

File: Doc/library/unittest.rst [hunk 2]
```diff
@@ -1571,6 +1571,12 @@ Test cases
 
    .. versionadded:: 3.8
 
+   .. attribute:: loop_factory
+
+      The *loop_factory* passed to :class:`asyncio.Runner` override
+      in subclasses with :class:`asyncio.EventLoop` to avoid using the
+      asyncio policy system.
+
```
Comment (graingert) [inline Doc/library/unittest.rst:1579]:
```suggestion
      asyncio policy system.
      
      .. versionadded:: 3.13
```

File: Doc/library/unittest.rst [hunk 3]
```diff
@@ -1571,6 +1571,13 @@ Test cases
 
    .. versionadded:: 3.8
 
+   .. attribute:: loop_factory
+
+      The *loop_factory* passed to :class:`asyncio.Runner` override
+      in subclasses with :class:`asyncio.EventLoop` to avoid using the
+      asyncio policy system.
+      
+      .. versionadded:: 3.13
```
Comment (graingert) [inline Doc/library/unittest.rst:1580]:
```suggestion
      .. versionadded:: 3.13

```

File: Doc/library/unittest.rst [hunk 4]
```diff
@@ -1571,6 +1571,14 @@ Test cases
 
    .. versionadded:: 3.8
 
+   .. attribute:: loop_factory
+
+      The *loop_factory* passed to :class:`asyncio.Runner` override
```
Comment (graingert) [inline Doc/library/unittest.rst]:
```suggestion
      The *loop_factory* passed to :class:`asyncio.Runner`. Override
```

File: Lib/test/test_unittest/test_async_case.py [hunk 5]
```diff
@@ -484,5 +484,19 @@ async def test_demo1(self):
         result = test.run()
         self.assertTrue(result.wasSuccessful())
 
+    def test_loop_factory(self):
+        asyncio.set_event_loop_policy(None)
+
+        class TestCase1(unittest.IsolatedAsyncioTestCase):
+            loop_factory = asyncio.EvetLoop
```
Comment (graingert) [inline Lib/test/test_unittest/test_async_case.py]:
```suggestion
            loop_factory = asyncio.EventLoop
```

**Review by willingc — APPROVED** [review]
Thanks @graingert. This looks good to me.

**Review by gvanrossum — APPROVED** [review]
Yup, thanks!

**Pull request merged**